libhwmon/wrapadl.cpp

```diff
@@ -192,12 +255,12 @@ int wrap_adl_get_tempC(wrap_adl_handle *adlh, int gpuindex, unsigned int *tempC)
 int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanpcnt)
 {
 	wrap_adlReturn_t rc;
-	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount)
+	if (gpuindex < 0 || gpuindex >= adlh->log_gpucount){
```

Comment:
please use adl_gpucount not log_gpucount. the adl count contain only physical devices and not all logical devices which can be a LOT.  All other wrappers use the physical index so if you have 5 cards index 0-4 works.
If we use logical 0-50 or so would be valid indexes, but get the same values till we are on the next physical device. like 0-7 or so is first card...8-14 the next.
The logical count also varies per physical card. it depends on its connectors and stuff.